File: cses/introductory/two_knights.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	fmt.Println(0)
	for k := 2; k < n+1; k++ {
		// we kinda divide both by 2 both due to double counting
		total := k * k * ((k * k) - 1) / 2
		attacks := 4 * 2 * (k - 1) * (k - 2) / 2
		fmt.Println(total - attacks)
	}
}

File: cses/introductory/two_knights.rs
use std::io;

fn main() {
    let mut n = String::new();
    io::stdin().read_line(&mut n).expect("Failed to read input");
    let n: usize = n.trim().parse().expect("Invalid input");

    println!("0");

    for k in 2..=n {
        // We divide both by 2 due to double counting
        let total = k * k * ((k * k) - 1) / 2;
        let attacks = 4 * 2 * (k - 1) * (k - 2) / 2;
        println!("{}", total - attacks);
    }
}
